crates/bevy_scene/src/bundle.rs

```diff
@@ -0,0 +1,94 @@
+use std::ops::{Deref, DerefMut};
+
+use bevy_asset::Handle;
+use bevy_ecs::{
+    bundle::Bundle,
+    change_detection::ResMut,
+    entity::Entity,
+    prelude::{Changed, Component},
+    system::{Commands, Query},
+};
+use bevy_transform::components::{GlobalTransform, Transform};
+
+use crate::{DynamicScene, InstanceId, Scene, SceneSpawner};
+
+/// [`InstanceId`] of a spawned scene. It can be used with the [`SceneSpawner`] to
+/// interact with the spawned scene.
+#[derive(Component)]
+pub struct SceneInstance(InstanceId);
+
+impl Deref for SceneInstance {
```

Comment:
This should be replaced with the new Deref and DerefMut derives.

Reply:
I think I disagree:
* Deref/DerefMut derives are pretty "small" and uncomplex. I highly doubt they meaningfully affect compile times
* I think this falls under the "engine code should look like user code" category. If we aren't willing to use our abstractions in internals, we shouldn't be recommending them to users (not a hard and fast rule, but a good rule of thumb imo).
* I don't think adding a dependency on bevy_derive meaningfully affects our dependency tree here.

Reply:
Also the less code we manually write out, the better.

Reply:
done